File: backend/models/ai_planner.py
from pydantic import BaseModel, Field
from typing import List, Optional, Dict, Any
from datetime import datetime

class WorkoutPlanRequest(BaseModel):
    """Model for workout plan request"""
    fitness_level: str = Field(..., description="User's fitness level (beginner, intermediate, advanced)")
    goals: List[str] = Field(..., description="User's fitness goals")
    available_equipment: Optional[List[str]] = Field(default=[], description="Available equipment for workouts")
    workout_days_per_week: int = Field(..., description="Number of days per week available for workouts")
    time_per_session: int = Field(..., description="Available time per workout session in minutes")
    preferences: Optional[List[str]] = Field(default=[], description="User's workout preferences")
    limitations: Optional[List[str]] = Field(default=[], description="User's physical limitations or injuries")

class Exercise(BaseModel):
    """Model for a single exercise"""
    name: str
    sets: int
    reps: str  # Can be a range like "8-12" or specific like "10"
    rest_time: str  # In seconds, e.g., "60s"
    description: Optional[str] = None
    equipment: Optional[List[str]] = None
    muscle_groups: List[str]
    difficulty: str  # beginner, intermediate, advanced
    instructions: Optional[str] = None
    alternatives: Optional[List[str]] = None

class WorkoutDay(BaseModel):
    """Model for a single workout day"""
    day: str  # e.g., "Day 1", "Monday", etc.
    focus: str  # e.g., "Upper Body", "Legs", "Full Body", etc.
    exercises: List[Exercise]
    warm_up: Optional[str] = None
    cool_down: Optional[str] = None
    total_time: Optional[int] = None  # Estimated time in minutes

class WorkoutPlan(BaseModel):
    """Model for a complete workout plan"""
    id: Optional[str] = None
    user_id: Optional[str] = None
    title: str
    description: str
    fitness_level: str
    goals: List[str]
    workout_days: List[WorkoutDay]
    created_at: Optional[datetime] = None
    notes: Optional[str] = None
    metadata: Optional[Dict[str, Any]] = None

class WorkoutPlanResponse(BaseModel):
    """Model for workout plan response"""
    plan: WorkoutPlan
    message: Optional[str] = None

File: src/models/AIPlanner.ts

// Original productivity planner interfaces (keeping for backward compatibility)
export interface UserPreferences {
  wakeUpTime: string;
  sleepTime: string;
  focusPeriods: number;
  breakDuration: number;
  primaryGoal: string;
}

export interface AIGeneratedPlan {
  dailySchedule: ScheduleItem[];
  weeklyFocus: string[];
  suggestedHabits: string[];
}

export interface ScheduleItem {
  id: string;
  time: string;
  activity: string;
  duration: number;
  priority: 'high' | 'medium' | 'low';
}

// Workout planner interfaces (matching backend models)
export interface WorkoutPlanRequest {
  fitness_level: string;
  goals: string[];
  available_equipment?: string[];
  workout_days_per_week: number;
  time_per_session: number;
  preferences?: string[];
  limitations?: string[];
}

export interface Exercise {
  name: string;
  sets: number;
  reps: string;
  rest_time: string;
  description?: string;
  equipment?: string[];
  muscle_groups: string[];
  difficulty: string;
  instructions?: string;
  alternatives?: string[];
}

export interface WorkoutDay {
  day: string;
  focus: string;
  exercises: Exercise[];
  warm_up?: string;
  cool_down?: string;
  total_time?: number;
}

export interface WorkoutPlan {
  id?: string;
  user_id?: string;
  title: string;
  description: string;
  fitness_level: string;
  goals: string[];
  workout_days: WorkoutDay[];
  created_at?: string;
  notes?: string;
  metadata?: Record<string, any>;
}

export interface WorkoutPlanResponse {
  plan: WorkoutPlan;
  message?: string;
}
